### PR TITLE
Eliminate duplicated conjunctive terms

### PR DESCRIPTION
**Pull request by b-scholz**
Fix issue #1170.

### CODE REVIEW

File: src/RamCondition.h [hunk 1]
```diff
@@ -492,4 +492,20 @@ inline std::unique_ptr<RamCondition> toCondition(const std::vector<const RamCond
...
 
+/**
+ * @brief Convert list of conditions to a conjunction
+ * @param A list of RAM conditions
+ * @param A RAM condition
+ *
+ * Convert a list {C1, C2, ..., Cn} to a condition of
+ * the format C1 /\ C2 /\ ... /\ Cn.
+ */
+inline std::unique_ptr<RamCondition> toCondition(const std::vector<std::unique_ptr<RamCondition>>& list) {
```
Comment (mmcgr) [inline src/RamCondition.h]: Can this function be replaced with `toCondition(toConstPtrVector(conds))` in RamTransforms.cpp (as done in Synthesiser.cpp)?

It might be better to redesign `toConjuctionList`/`toCondition`, but that would be a separate PR.

Comment (b-scholz) [reply src/RamCondition.h]: Yes, indeed. We should raise this as an issue and prepare a later PR to sort this inconsistency out.

File: src/main.cpp [hunk 2]
```diff
@@ -489,8 +489,9 @@ int main(int argc, char** argv) {
             std::make_unique<RamLoopTransformer>(std::make_unique<RamTransformerSequence>(
                     std::make_unique<HoistAggregateTransformer>(), std::make_unique<TupleIdTransformer>())),
             std::make_unique<ExpandFilterTransformer>(), std::make_unique<HoistConditionsTransformer>(),
-            std::make_unique<CollapseFiltersTransformer>(), std::make_unique<ReorderConditionsTransformer>(),
-            std::make_unique<RamLoopTransformer>(std::make_unique<ReorderFilterBreak>()),
```
Comment (XiaowenHu96) [inline src/main.cpp]: Why is `ReorderFilterBreak` removed from the pipeline?

Comment (b-scholz) [reply src/main.cpp]: Indeed - this needs to be fixed.
